frontend/javascripts/test/puppeteer/dtype_datasets_rendering.screenshot.ts

```diff
@@ -0,0 +1,322 @@
+  alsoTestSelectiveSegmentId?: boolean;
+};
+
+const specs: Array<Spec> = _.flatten(
+  dtypes.map((dtype): Spec[] => {
+    const elementClass = dtype === "float32" ? "float" : dtype;
+
+    // No color support for 64 bit
+    const colorSpecs = ["uint64", "int64"].includes(elementClass)
+      ? []
+      : [
+          {
+            name: `dtype_${dtype}_color_${zoomedIn.postfix}`,
+            dtype,
+            datasetName: `dtype_test_${dtype}_color`,
+            viewOverride: zoomedIn.viewOverride,
+            datasetConfig: datasetConfigHelper(`${dtype}_color`, [
+              getDtypeConfigForElementClass(elementClass).isSigned ? -10 : 0,
+              10,
+            ]),
+          },
+          {
+            name: `dtype_${dtype}_color_${zoomedOut.postfix}`,
+            dtype,
+            datasetName: `dtype_test_${dtype}_color`,
+            viewOverride: zoomedOut.viewOverride,
+            datasetConfig: datasetConfigHelper(
+              `${dtype}_color`,
+              getSupportedValueRangeForElementClass(elementClass),
+            ),
+          },
+        ];
+
+    const segmentationSpecs =
+      // No segmentation support for float
+      elementClass === "float"
+        ? []
+        : [
+            {
+              name: `dtype_${dtype}_segmentation_${zoomedIn.postfix}`,
+              dtype,
+              datasetName: `dtype_test_${dtype}_segmentation`,
+              viewOverride: zoomedIn.viewOverride,
+              datasetConfig: datasetConfigHelper(`${dtype}_segmentation`, undefined),
+              alsoTestSelectiveSegmentId: true,
+            },
+            {
+              name: `dtype_${dtype}_segmentation_${zoomedOut.postfix}`,
+              dtype,
+              datasetName: `dtype_test_${dtype}_segmentation`,
+              viewOverride: zoomedOut.viewOverride,
+              datasetConfig: datasetConfigHelper(`${dtype}_segmentation`, undefined),
+            },
+          ];
+
+    if (testSegmentation && testColor) {
+      return [...colorSpecs, ...segmentationSpecs];
+    } else if (testSegmentation) {
+      return segmentationSpecs;
+    } else if (testColor) {
+      return colorSpecs;
+    } else {
+      throw new Error("Both testColor and testSegmentation are set to false.");
+    }
+  }),
+);
+
+const datasetNames = _.uniq(specs.map((spec) => spec.datasetName));
+
+const datasetNameToId: Record<string, string> = {};
+test.before("Retrieve dataset ids", async () => {
+  for (const datasetName of datasetNames) {
+    await withRetry(
+      1,
+      async () => {
+        const options = getDefaultRequestOptions(URL);
+        const path = `/api/datasets/disambiguate/sample_organization/${datasetName}/toId`;
+        const url = urljoin(URL, path);
+        const response = await fetch(url, options);
+        const { id } = await response.json();
+        datasetNameToId[datasetName] = id;
+        return true;
+      },
+      () => {},
+    );
+  }
+});
```

Comment:
This is a complete duplication from the other `screenshot.ts` file. Could you please deduplicate this?